src/components/views/auth/CaptchaForm.tsx

```diff
@@ -15,66 +15,74 @@ limitations under the License.
 */
 
 import React, { createRef } from 'react';
-import PropTypes from 'prop-types';
 import { _t } from '../../../languageHandler';
 import CountlyAnalytics from "../../../CountlyAnalytics";
 import { replaceableComponent } from "../../../utils/replaceableComponent";
 
 const DIV_ID = 'mx_recaptcha';
 
+interface ICaptchaFormProps {
+    sitePublicKey: string;
+    onCaptchaResponse: (response: string) => void;
+}
+
+interface ICaptchaFormState {
+    errorText: string | null;
+
+}
+
 /**
  * A pure UI component which displays a captcha form.
  */
 @replaceableComponent("views.auth.CaptchaForm")
-export default class CaptchaForm extends React.Component {
-    static propTypes = {
-        sitePublicKey: PropTypes.string,
-
-        // called with the captcha response
-        onCaptchaResponse: PropTypes.func,
-    };
-
+export default class CaptchaForm extends React.Component<ICaptchaFormProps, ICaptchaFormState> {
     static defaultProps = {
         onCaptchaResponse: () => {},
     };
 
-    constructor(props) {
+    private captchaWidgetId: string | null = null;
```

Comment:
```suggestion
    private captchaWidgetId: string | null = null; 
```

We don't tend to specify `| null` here given the `= null` makes that implicit

Reply:
@t3chguy it doesn't, it works like in Java without specifying the null and expects that it will always be string.
This should throw in the compiler if it were to be implicitly defined as a null, yet it doesn't:
```

        const z = this.captchaWidgetId
        function test(a: string) {
            console.log(a)
        }
        test(z)
```
The only way to properly catch nulls and make them not behave like in Java (where compiler just doesn't know and assumes it's your defined type, so you have to remember about explicitly checking it) would be to add `strictNullChecks` to `tsconfig.json`, but then you'd have to define the `null` type anyways.

Please let me know your opinion of that after reading my comment so that I can make it proper according to the styleguide, but I don't necessarily agree with it. 😆